File: internal/entry/time.go
package entry

import (
	"fmt"
	"time"
)

func MakeTimestamp() string {
	now := time.Now()

	// format: yyyy-mm-dd-hh
	timestamp := fmt.Sprintf(
		"%04d-%02d-%02d-%02d-%02d-%02d",
		now.Year(),
		int(now.Month()),
		now.Day(),
		now.Hour(),
		now.Minute(),
		now.Second(),
	)

	return timestamp
}

func GetYearMonth() string {
	now := time.Now()

	timestamp := fmt.Sprintf(
		"%004d-%02d",
		now.Year(),
		int(now.Month()),
	)

	return timestamp
}

File: docs/src/utils/time.js
export function makeTimeString(timeStr) {
	return timeAgo(timeStr);
}

export function timeAgo(timestamp) {
	const iso = timestamp.replace(
		/^(\d{4}-\d{2}-\d{2})-(\d{2})-(\d{2})-(\d{2})$/,
		"$1T$2:$3:$4"
	);

	const past = new Date(iso);
	const now = new Date();

	const diffMs = now - past;
	if (isNaN(diffMs) || diffMs < 0) return "invalid date";

	const seconds = Math.floor(diffMs / 1000);
	if (seconds < 60) return "just now";

	const minutes = Math.floor(seconds / 60);
	if (minutes < 60) {
		return `${minutes} minute${minutes === 1 ? "" : "s"} ago`;
	}

	const hours = Math.floor(minutes / 60);
	if (hours < 24) {
		return `${hours} hour${hours === 1 ? "" : "s"} ago`;
	}

	const days = Math.floor(hours / 24);
	if (days < 30) {
		return `${days} day${days === 1 ? "" : "s"} ago`;
	}

	const months = Math.floor(days / 30);
	if (months < 12) {
		return `${months} month${months === 1 ? "" : "s"} ago`;
	}

	const years = Math.floor(months / 12);
	return `${years} year${years === 1 ? "" : "s"} ago`;
}

export function makeTitleTimestamp(timestamp) {
	const [yyyy, Mm, dd, hh, mm, ss] = timestamp.split("-");
	return `${yyyy}-${Mm}-${dd} ${hh}:${mm}:${ss}`;
}

